test/exqlite/sqlite3_test.exs

```diff
@@ -491,4 +491,68 @@ defmodule Exqlite.Sqlite3Test do
       refute_receive {{:insert, "main", "test", 1}, _}, 1000
     end
   end
+
+  describe "set_log_hook/2" do
+    setup do
+      {:ok, path} = Temp.path()
+      {:ok, conn} = Sqlite3.open(path)
+
+      on_exit(fn ->
+        Sqlite3.close(conn)
+        File.rm(path)
+      end)
+
+      {:ok, conn: conn, path: path}
+    end
+
+    test "can receive errors", %{conn: conn} do
+      assert :ok = Sqlite3.set_log_hook(conn, self())
+
+      assert {:error, reason} = Sqlite3.prepare(conn, "some invalid sql")
+      assert reason == "near \"some\": syntax error"
+
+      assert_receive {:log, rc, msg}
+      assert rc == 1
+      assert msg == "near \"some\": syntax error in \"some invalid sql\""
+      refute_receive _anything_else
+    end
+
+    test "only one pid can listen at a time", %{conn: conn} do
+      test = self()
+      assert :ok = Sqlite3.set_log_hook(conn, test)
+
+      spawn(fn ->
+        :ok = Sqlite3.set_log_hook(conn, self())
+        assert {:error, reason} = Sqlite3.prepare(conn, "some invalid sql")
+        assert reason == "near \"some\": syntax error"
+        assert_receive {:log, rc, msg}
+        assert rc == 1
+        assert msg == "near \"some\": syntax error in \"some invalid sql\""
+        refute_receive _anything_else
+        send(test, :done)
+      end)
+
+      assert_receive :done, 1000
+      refute_receive _anything_else
+    end
+
+    test "notifications don't cross connections", %{conn: conn1, path: path} do
```

Comment:
I wonder how this works though, since the `sqlite_config` call seems to be global as it doesn't require a DB reference in its args.